pkg/agentfs/docker.go

```diff
@@ -51,94 +54,249 @@ func CreateDockerfile(dir string, settingsMap map[string]string) error {
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to write .dockerignore: %w", err)
 	}
 
+	fmt.Printf("\n✔ Successfully generated Docker files:\n")
+	fmt.Printf("  %s - Container build instructions\n", util.Accented("Dockerfile"))
+	fmt.Printf("  %s - Files excluded from build context\n", util.Accented(".dockerignore"))
+	fmt.Printf("\nNext steps:\n")
+	fmt.Printf("  ► Review the %s and uncomment any needed system packages\n", util.Accented("Dockerfile"))
+	fmt.Printf("  ► Build your agent: docker build -t my-agent .\n")
+	fmt.Printf("  ► Test locally: docker run my-agent\n")
+
 	return nil
 }
 
-func validateEntrypoint(dir string, dockerfileContent []byte, projectType string, settingsMap map[string]string) ([]byte, error) {
-	fileList := make(map[string]bool)
-	entries, err := os.ReadDir(dir)
+func validateUVProject(dir string) {
+	uvLockPath := filepath.Join(dir, "uv.lock")
+	if _, err := os.Stat(uvLockPath); err != nil {
+		fmt.Printf("! Warning: UV project detected but %s file not found\n", util.Accented("uv.lock"))
+		fmt.Printf("  Consider running %s to generate %s for reproducible builds\n", util.Accented("uv lock"), util.Accented("uv.lock"))
+		fmt.Printf("  This ensures consistent dependency versions across environments\n\n")
+	}
+}
+
+func validateEntrypoint(dir string, dockerfileContent []byte, dockerignoreContent []byte, projectType ProjectType, settingsMap map[string]string) ([]byte, error) {
+	// Parse dockerignore patterns to filter out files that won't be in build context
+	reader := bytes.NewReader(dockerignoreContent)
+	patterns, err := ignorefile.ReadAll(reader)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to parse .dockerignore: %w", err)
+	}
+	matcher, err := patternmatcher.New(patterns)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create pattern matcher: %w", err)
 	}
 
-	for _, entry := range entries {
-		if entry.IsDir() {
-			continue
+	// Use recursive traversal to find all relevant files, respecting dockerignore
+	var allFiles []string
+	if err := filepath.WalkDir(dir, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip files that match .dockerignore patterns
+		if ignored, err := matcher.MatchesOrParentMatches(path); ignored {
+			return nil
+		} else if err != nil {
+			return err
 		}
-		fileList[entry.Name()] = true
+
+		// Only include files with the correct extension
+		if !d.IsDir() && strings.HasSuffix(d.Name(), projectType.FileExt()) {
+			// Convert to relative path from directory
+			relPath, err := filepath.Rel(dir, path)
+			if err != nil {
+				return err
+			}
+			allFiles = append(allFiles, relPath)
+		}
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("error walking directory %s: %w", dir, err)
+	}
+
+	// Convert to map for our existing prioritization logic compatibility
+	fileList := make(map[string]bool)
+	for _, file := range allFiles {
+		fileList[file] = true
 	}
```

Comment:
If we're gonna use a map, let's just use one from the start. Also, the "map of empty structs as a set" pattern is a tiny but very "go-pilled" optimization here 😛 

```go
allFiles := make(map[string]struct{})
```